flang/lib/Lower/IO.cpp

```diff
@@ -941,6 +941,25 @@ mlir::Value genIOOption<Fortran::parser::ConnectSpec::Recl>(
   return genIntIOOption<mkIOKey(SetRecl)>(converter, loc, cookie, spec);
 }
 
+template <>
+mlir::Value genIOOption<Fortran::parser::ConnectSpec::Newunit>(
+    Fortran::lower::AbstractConverter &converter, mlir::Location loc,
+    mlir::Value cookie, const Fortran::parser::ConnectSpec::Newunit &spec) {
+  Fortran::lower::StatementContext stmtCtx;
+  fir::FirOpBuilder &builder = converter.getFirOpBuilder();
+  mlir::FuncOp ioFunc = getIORuntimeFunc<mkIOKey(GetNewUnit)>(loc, builder);
+  mlir::FunctionType ioFuncTy = ioFunc.getType();
+  const auto *var = Fortran::semantics::GetExpr(spec);
+  mlir::Value addr = builder.createConvert(
+      loc, ioFuncTy.getInput(1),
+      fir::getBase(converter.genExprAddr(var, stmtCtx, loc)));
+  auto kind = builder.create<mlir::arith::ConstantOp>(
```

Comment:
Using `builder.createConstantInteger` here would hide the attribute handling part.

Reply:
That's better code; thanks